Framework/Core/include/Framework/DeviceMetricsInfo.h

```diff
@@ -80,6 +86,7 @@ struct DeviceMetricsInfo {
   std::vector<size_t> maxDomain;
   std::vector<MetricLabelIndex> metricLabelsIdx;
   std::vector<MetricInfo> metrics;
+  std::string deviceName;
```

Comment:
Why do you need to copy the device name?

Reply:
Ok, I think I see. Please do not copy the information, simply retrieve it from the DeviceSpec when you dump it.